Clustering/Hdbscan/src/main/java/org/tribuo/clustering/hdbscan/HdbscanModel.java

```diff
@@ -115,18 +118,38 @@
         if (vector.numActiveElements() == 0) {
             throw new IllegalArgumentException("No features found in Example " + example);
         }
+
         double minDistance = Double.POSITIVE_INFINITY;
-        int clusterLabel = -1;
-        double clusterOutlierScore = 0.0;
-        for (HdbscanTrainer.ClusterExemplar clusterExemplar : clusterExemplars) {
-            double distance = DistanceType.getDistance(clusterExemplar.getFeatures(), vector, distType);
-            if (distance < minDistance) {
-                minDistance = distance;
-                clusterLabel = clusterExemplar.getLabel();
-                clusterOutlierScore = clusterExemplar.getOutlierScore();
+        int clusterLabel = HdbscanTrainer.OUTLIER_NOISE_CLUSTER_LABEL;
+        double outlierScore = 0.0;
+        if (Double.compare(noisePointsOutlierScore, 0) > 0) { // This will be true from models > 4.2
+            boolean isNoisePoint = true;
+            for (HdbscanTrainer.ClusterExemplar clusterExemplar : clusterExemplars) {
+                double distance = DistanceType.getDistance(clusterExemplar.getFeatures(), vector, distType);
+                if (isNoisePoint && distance <= clusterExemplar.getMaxDistToEdge()) {
+                    isNoisePoint = false;
+                }
+                if (distance < minDistance) {
+                    minDistance = distance;
+                    clusterLabel = clusterExemplar.getLabel();
+                    outlierScore = clusterExemplar.getOutlierScore();
+                }
+            }
+            if (isNoisePoint) {
+                outlierScore = noisePointsOutlierScore;
```

Comment:
This doesn't update the cluster label to the noise cluster? There's nothing in the if statement on distance to prevent it from setting the cluster, so I'm not sure if it will assign noise clusters at prediction time.

Reply:
Yes, you are right. I've corrected this and added some new assertions to an existing test to ensure noise points are being correctly predicted.